### PR TITLE
Add greeting to Account Balances header

### LINKED ISSUE #5721: Simplify deposits UI - Add greeting header panel to balances card
### Description

This issue is part of the larger Epic to add the Payments Overview Deposits card (https://github.com/Automattic/woocommerce-payments/issues/5581).

As part of this issue, the card should altered to include the following acceptance criteria:

### Acceptance criteria 

- Greeting. 
    - "Good afternoon, {user-first-name} 👋".
    - "Good morning, {user-first-name} 👋".
- Woo Payments graphic.

224232319-3dd3bb14-b0c8-469f-8fdc-8557fcc82199

### Designs

P2: paJDYF-7fi-p2

image

### Additional context

Project: pdjTHR-2i5-p2

### PR DESCRIPTION
**Pull request by Jinksi**
Resolves #5721

### Changes proposed in this Pull Request

The work in this PR is currently not user-facing, it will only be rendered if the `_wcpay_feature_simplify_deposits_ui` feature flag is enabled.

This PR adds a greeting to the Account Balances card header, based on the time of day and the currently logged in user's first name.

The new Woo Payments logo has been added to the right side of the header.

account-balances-header

**Greeting time of day**

The greeting will adjust to the following based on the time of day. Source for these times: pdjTHR-2jn-p2#comment-3031.

Good morning [5am -11.59am]
Good afternoon [12pm – 4:59pm]
Good evening [5pm – 4:59am]

**Greeting name**
The name will match the logged-in user's "First Name" from `/wp-admin/profile.php`. If the "First Name" field is empty, the greeting will fall back to "Good {time of day} 👋".

### Designs

Design P2: paJDYF-7fi-p2

Design for this card header component:

### Testing instructions

* Run `npm run build:client`.
* Enable the Simplify Deposits UI feature flag.  
  `update_option( '_wcpay_feature_simplify_deposits_ui', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/).
* Go to Payments → Overview.
* At the top of the Account Balances card, you should see the header with a greeting and Woo Payments logo that matches the design shown above.
* Greeting time of day: adjust your system's time (or wait) and ensure the correct time of day is displayed in the greeting, as per times in the description above.
* The greeting name should match the logged-in user's "First Name" from `/wp-admin/profile.php`.
* Set the logged-in user's "First Name" to an empty string via `/wp-admin/profile.php`. The greeting should fall back to "Good {time of day} 👋".
* The greeting should not visually break when provided a First Name with many characters, e.g. "Chrysanthemum".
* Ensure the header styles are responsive by reducing the viewport width or testing on a mobile device. Example of responsive styles on mobile with a long user "First Name":
   image
* Optional: Install the latest version of the Gutenberg plugin and ensure that the header styles look correct. (conflicting styles can occur with different versions of `@wordpress/components`)

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Todo:**
- [x] Add greeting.
- [x] Add Woo Payments logo.
- [x] Lock-in vertical alignment across header greeting and logo
- [x] Make responsive.

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/components/account-balances/utils.ts
Comment (Jinksi) [inline client/components/account-balances/utils.ts]: Source for these times:  pdjTHR-2jn-p2#comment-3031

File: client/components/account-balances/header.tsx
Comment (Jinksi) [inline client/components/account-balances/header.tsx:12]: This import target ~will~ can change from `../../../assets/` to `assets/` once #5609 is merged.

File: assets/images/woopayments.svg
Comment (Jinksi) [inline assets/images/woopayments.svg:1]: I've optimised this SVG file as per instructions in #5609.

File: client/components/account-balances/utils.ts
Comment (james-allan) [inline client/components/account-balances/utils.ts:49]: Just a pre-submittal heads up. When translating strings they should be translated as a complete sentence. 

ie 
```php
// Placeholder is user's name. 
__( 'Good evening, %s', 'woocommerce-payments' );
```


This is because some languages need to change the sentence structure. eg in Polish it's "dzień dobry Jamesie" (Day good, James) or "Good" on it's own may be translated into 1 word but a different word in this context (greeting).

Comment (james-allan) [reply client/components/account-balances/utils.ts:49]: This somewhat complicates things because you've got the user's name being conditionally appended. 

But you could change the `timeOfDayStrings` into full greetings. eg `Good morning, %', 'Good afternoon, %s', 'Good evening, %s'.

and if conditional user names are something we need to account for, I can only think we then need another set of greetings without username placeholds. 🙈

Comment (Jinksi) [reply client/components/account-balances/utils.ts:49]: Thanks for the heads up @james-allan 🙌

Comment (Jinksi) [reply client/components/account-balances/utils.ts:49]: I've changed these strings to remain as entire phrases in ef916e4

File: client/components/account-balances/header.tsx
Comment (james-allan) [reply client/components/account-balances/header.tsx:12]: If we merge this before that PR I'm assuming we'll need to either update #5609 or come back and tidy up?

File: client/components/account-balances/utils.ts
Comment (james-allan) [inline client/components/account-balances/utils.ts]:
```suggestion
```

There's an extra line here.

File: client/components/account-balances/utils.ts
Comment (james-allan) [inline client/components/account-balances/utils.ts]:
```suggestion
 * @param {string} name A name to include in the greeting, optional.
```

Small typo :)

File: client/components/account-balances/header.tsx
Comment (Jinksi) [reply client/components/account-balances/header.tsx:12]: Yeh, a bit of a race condition happening here.

This will work regardless and will have no impact on project outcomes.

I would categorise this as a minor code quality consideration only.

**Review by james-allan — APPROVED** [review]
This looks good to me. 

I tested changing my times to the different windows for evening, afternoon and morning. 

I tried testing the no name scenario however it always fell back to the user's nickname and the nickname cannot be empty. Even when I went into the database and deleted it, it would regenerate. 

I think this is fine, there may be some redundancy in the no name code etc but I think that's fine. 

I just left 2 minor CR things I came across. :)

File: client/components/account-balances/hooks.ts
Comment (james-allan) [inline client/components/account-balances/hooks.ts]: Do you happen to know what the user.name property is here?

Theres a username, nick name, first name and name. 🤔

Comment (Jinksi) [reply client/components/account-balances/hooks.ts]: `user.name` is the "Display name publicly as" value from `wp-admin/profile.php`.

Thinking more about the intention of this greeting, I think it is best to limit this to only return "First Name". The usage of `user.nickname` and `user.name` is likely to give undesirable results.

Comment (Jinksi) [reply client/components/account-balances/hooks.ts]: > I think it is best to limit this to only return "First Name"

I've made this change in dd78aaa.

Comment (james-allan) [reply client/components/account-balances/hooks.ts]: > Thinking more about the intention of this greeting, I think it is best to limit this to only return "First Name". The usage of user.nickname and user.name is likely to give undesirable results.

Yeah good point. When I went and deleted my first name it came up with "Good morning, james 👋" - uncapitalised j. I can just imagine this will grind on peoples gears seeing their name incorrectly capitalised.

**Comment by james-allan** [comment]
The changes since my last approval look good to me. 👍

File: client/components/account-balances/header.tsx
Comment (ismaeldcom) [reply client/components/account-balances/header.tsx:12]: I've handled it as part of the merge with `develop`, thanks for noticing it! 🙂

**Pull request merged**
